src/xb-common.c

```diff
@@ -96,3 +97,44 @@ xb_content_type_guess (const gchar *filename, const guchar *buf, gsize bufsz)
 
 	return g_steal_pointer (&content_type);
 }
+
+/**
+ * xb_file_set_contents: (skip)
+ * @file: (nullable): file to write
+ * @buf: (nullable): data buffer
+ * @bufsz: size of @buf
+ * @cancellable: (nullable): optional #GCancellable
+ * @error: (nullable): optional return location for an error
+ *
+ * Writes data to a file. This is done safely using a temporary file (and the
+ * using rename() to do this atomically) on Linux but a direct write is used
+ * on Windows.
+ *
+ * Returns: a string indicating a guessed content type
+ **/
+gboolean
+xb_file_set_contents (GFile *file, const guint8 *buf, gsize bufsz,
+		      GCancellable *cancellable,
+		      GError **error)
+{
+#ifdef _WIN32
+	g_autofree gchar *fn = g_file_get_path (file);
+#endif
+
+	g_return_val_if_fail (G_IS_FILE (file), FALSE);
+	g_return_val_if_fail (buf != NULL, FALSE);
+	g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+#if defined(_WIN32) && GLIB_CHECK_VERSION(2,66,0)
+	return g_file_set_contents_full (fn, (const gchar *) buf, (gssize) bufsz,
+					 G_FILE_SET_CONTENTS_NONE, 0666, error);
```

Comment:
so what about WIN32 and older glib though?  Does that mean it will fail there?  Should it be an error?

Reply:
Yes, you're right. Fixed!